runtime/src/portablegabi.rs

```diff
@@ -0,0 +1,148 @@
+use support::{
+	decl_event, decl_module, decl_storage, ensure, dispatch::Result, StorageMap,
+};
+use rstd::vec::Vec;
+use system::ensure_signed;
+
+/// The pallet's configuration trait.
+pub trait Trait: system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as TemplateModule {
+		/// The AccumulatorList contains all accumulator. It is a map which
+		/// maps an account id and an index to an accumulator
+		AccumulatorList get(accumulator_list): map (T::AccountId, u64) => Option<Vec<u8>>;
+
+		/// The AccumulatorCounter stores for each attester the number of
+		/// accumulator updates.
+		AccumulatorCount get(accumulator_count): map T::AccountId => u64;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		// Initializing events
+		fn deposit_event<T>() = default;
+
+		/// Updates the attestation
+		pub fn update_accumulator(origin, accumulator: Vec<u8>) -> Result {
+			let attester = ensure_signed(origin)?;
+
+			let counter = if !<AccumulatorCount<T>>::exists(&attester) {
+				0
+			} else {
+				<AccumulatorCount<T>>::get(&attester)
+			};
+
+			let next = counter.checked_add(1).ok_or("Overflow increasing accumulator index")?;
+			ensure!(!<AccumulatorList<T>>::exists((attester.clone(), next)),
+					"Inconsistent accumulator counter");
```

Comment:
Please use error module

Reply:
fixed in cb2ed89659a451adee7333d9459bcbb2fe6a8ee1